Add flux dockerfile

SteVwonder opened the pull request:
Builds on top of #16.  It was only after trying to compile Flux that I realized the `python36-cffi` package is only in EPEL.  So this PR adds EPEL before installing all of the dependency packages.  Otherwise, the previous `RUN` commands from #16 should be the same.

This PR ensures that flux is built correctly by running `make check` as part of the image build.  It also adds a small test.sh that ensures flux starts and can run a simple single-core job.

While the Dockerfile is technically parameterized to support different versions of flux-core, I only tested that the latest release (0.24.0) builds and runs successfully.

Closes #4

dongahn commented:
From our today's discussion, @SteVwonder will add a few modification and force a push.

dongahn commented on docker/flux/Dockerfile:
```diff
@@ -0,0 +1,35 @@
+   && ldconfig
+
+COPY test.sh /
+CMD /test.sh
```
Super minor nit. My web browser flag this line as not having the new line character. This is a false poise but it would be good to add a newline so that others with the same renderer as mine don't have to pay attention.

dongahn reviewed (approved):
This looks great to me @SteVwonder. I just added a minor comment. I approve this PR so feel free to merge it if/when you add a new line to a commit. Thanks.

dongahn commented:
FYI -- need to be rebased before going in.

SteVwonder commented:
Thanks @dongahn .  Added a newline to the docker file, added the flux image to the CI, and rebase onto master.  Let's hope the CI tests still pass after the new addition. 🤞

dongahn commented:
Great. Merging.
